imap/index_file.c

```diff
@@ -0,0 +1,580 @@
+    {  4, '4', offsetof(struct index_header, recentuid)            },
+    {  4, 'T', offsetof(struct index_header, recenttime)           },
+    {  4, 'T', offsetof(struct index_header, pop3_show_after)      },
+    {  4, '8', offsetof(struct index_header, quota_annot_used)     },
+    {  4, '4', offsetof(struct index_header, synccrcs.annot)       },
+    {  4, '4', offsetof(struct index_header, unseen)               },
+    {  8, '8', offsetof(struct index_header, createdmodseq)        },
+    {  8, '8', offsetof(struct index_header, quota_deleted_used)   },
+    {  8, '8', offsetof(struct index_header, quota_expunged_used)  },
+    {  4, 'T', offsetof(struct index_header, changes_epoch)        },
+    NULL_FIELD
+};
+
+
+/* v20 grew size of time fields and quota_annot_used to 64-bits
+   and rearranged fields so that these would fall on 8-byte boundaries */
+static const index_field_t v20_header_fields[] = {
+    {  8, 'T', offsetof(struct index_header, last_appenddate)      },
+    {  8, '8', offsetof(struct index_header, quota_mailbox_used)   },
+    {  8, 'T', offsetof(struct index_header, pop3_last_login)      },
+    {  4, '4', offsetof(struct index_header, deleted)              },
+    {  4, '4', offsetof(struct index_header, answered)             },
+    {  4, '4', offsetof(struct index_header, flagged)              },
+    {  4, '4', offsetof(struct index_header, exists)               },
+    {  4, '4', offsetof(struct index_header, options)              },
+    {  4, '4', offsetof(struct index_header, leaked_cache_records) },
+    {  8, '8', offsetof(struct index_header, highestmodseq)        },
+    {  8, '8', offsetof(struct index_header, deletedmodseq)        },
+    {  4, '4', offsetof(struct index_header, last_uid)             },
+    {  4, '4', offsetof(struct index_header, uidvalidity)          },
+    {  4, '4', offsetof(struct index_header, header_file_crc)      },
+    {  4, '4', offsetof(struct index_header, synccrcs.basic)       },
+    {  8, 'T', offsetof(struct index_header, recenttime)           },
+    {  8, 'T', offsetof(struct index_header, pop3_show_after)      },
+    {  4, '4', offsetof(struct index_header, synccrcs.annot)       },
+    {  4, '4', offsetof(struct index_header, unseen)               },
+    {  8, '8', offsetof(struct index_header, createdmodseq)        },
+    {  8, '4', offsetof(struct index_header, quota_deleted_used)   },
+    {  8, '8', offsetof(struct index_header, quota_expunged_used)  },
+    {  8, '8', offsetof(struct index_header, quota_annot_used)     },
+    {  8, 'T', offsetof(struct index_header, changes_epoch)        },
+    {  8, 'T', offsetof(struct index_header, first_expunged)       },
+    {  8, 'T', offsetof(struct index_header, last_repack_time)     },
+    {  4, '4', offsetof(struct index_header, recentuid)            },
+    NULL_FIELD
+};
+
+
+/*
+ * Index file record fields by version
+ *
+ * Comment at the end of each field is the offset of the field in record buffer
```

Comment:
I think this comment also doesn't make sense anymore?

Reply:
Fixed